warp10/src/main/java/io/warp10/continuum/Tokens.java

```diff
@@ -253,7 +253,28 @@ public static ReadToken extractReadToken(String token) throws WarpScriptExceptio
     
     List<String> owners = new ArrayList<String>();
     List<String> producers = new ArrayList<String>();
-    
+    Map<String, String> labels = new HashMap<String, String>();
+
+    if (rtoken.getLabelsSize() > 0) {
+      labels = rtoken.getLabels();
+      if (!labels.isEmpty()) {
+        for (Map.Entry<String, String> entry : labels.entrySet())
+        {
+          switch (entry.getKey())
+          {
+            case Constants.OWNER_LABEL:
+              continue;
```

Comment:
the first two continue statements could be removed with a fall through to the PRODUCER_LABEL case.